lib/components/SideMenu/side-menu.ts

```diff
@@ -113,7 +114,21 @@ export class SideMenu implements OnInit, OnDestroy {
 
   init() {
     this.menuItems = this.menuService.items;
-
+    let firstMenuItemsLength = this.menuItems.length;
+
+    for (var firstMenuItemIndex = 0; firstMenuItemIndex < firstMenuItemsLength; firstMenuItemIndex++) {
```

Comment:
Menu items can have more than 2 levels. Better to not hardcode two levels here but replace this with a recursive sorting algorithm which will support any nesting

Also line 124 may be executed multiple for each `firstMenuItemIndex` which is unnecessary work